src/plugins/query_enhancements/server/routes/index.ts

```diff
@@ -12,151 +12,130 @@ import {
 } from '../../../../core/server';
 import { IDataFrameResponse, IOpenSearchDashboardsSearchRequest } from '../../../data/common';
 import { ISearchStrategy } from '../../../data/server';
-import { API, SEARCH_STRATEGY } from '../../common';
+import { API } from '../../common';
 import { registerQueryAssistRoutes } from './query_assist';
 import { registerDataSourceConnectionsRoutes } from './data_source_connection';
 
-/**
- * Defines a route for a specific search strategy.
- *
- * @experimental This function is experimental and might change in future releases.
- *
- * @param logger - The logger instance.
- * @param router - The router instance.
- * @param searchStrategies - The available search strategies.
- * @param searchStrategyId - The ID of the search strategy to use.
- *
- * @example
- * API Request Body:
- * ```json
- * {
- *   "query": {
- *     "query": "SELECT * FROM my_index",
- *     "language": "sql",
- *     "dataset": {
- *       "id": "my_dataset_id",
- *       "title": "My Dataset"
- *     },
- *     "format": "json"
- *   },
- *   @experimental
- *   "aggConfig": {
- *     // Optional aggregation configuration
- *   },
- *   @deprecated
- *   "df": {
- *     // Optional data frame configuration
- *   }
- * }
- * ```
- */
-function defineRoute(
-  logger: Logger,
-  router: IRouter,
-  searchStrategies: Record<
-    string,
-    ISearchStrategy<IOpenSearchDashboardsSearchRequest, IDataFrameResponse>
-  >,
-  searchStrategyId: string
-) {
-  const path = `${API.SEARCH}/${searchStrategyId}`;
-  router.post(
-    {
-      path,
-      validate: {
-        body: schema.object({
-          query: schema.object({
-            query: schema.string(),
-            language: schema.string(),
-            dataset: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-            format: schema.string(),
+export function defineSearchStrategyRouteProvider(logger: Logger, router: IRouter) {
```

Comment:
Can we keep this experimental?

Reply:
You mean have a comment saying experimental?